packages/components/pie-modal/src/index.ts

```diff
@@ -328,15 +351,16 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
                 </${headingTag}>
                 ${isDismissible ? this.renderCloseButton() : nothing}
             </header>
-            <article class="c-modal-content c-modal-content--scrollable">
-                <div class="c-modal-contentInner">
-                    <slot></slot>
-                </div>
-            </article>
-            <footer class="c-modal-footer">
-                ${leadingAction ? this.renderLeadingAction() : nothing}
-                ${supportingAction ? this.renderSupportingAction() : nothing}
-            </footer>
+            ${
```

Comment:
You could potentionally shorten this and make use of the isFooterPinned prop to render a class or a generic class instead. That way you dont require the ternary and can have a shorter bit of code.
```html
<div class=${isFooterPinned ? "c-modal-footer-pinned" : "c-modal-scrollContainer"}>
  ${this.renderModalContentAndFooter()}
</div>

Reply:
I was trying to avoid a needless div! What would `c-modal-footer-pinned` be used for in this case?

Reply:
Fair enough. More of a suggestion. I think the bigger this component gets the harder it is going to be to follow / read. Maybe even moving it out to a separate method would help and then rendering that method as a single thing here would just make it clearer.

Reply:
I do agree, but I'm unsure if adding another layer of abstraction (i.e. another method) would add any coherence here. I'd prefer to go with this for now and optimize once all the markup is added to the modal and we have a clearer understanding of the final 'shape' of the html